### DISCUSSION TITLE
Get output parameter from Stored Procedure?

### DISCUSSION THREAD

**Discussion by Greatchinex** [SeaQL/sea-orm#1924] [Q&A]
Hello, Please i am very new to rust and i want to use stored procedures for saving transaction and updating wallet balance, But i am just test running a simple procedure to create a user first. But i can't get the output from the procedure. I have something like

```
let email = "john@john.com"; let uuid = Uuid::new_v4().to_string();

let exec = db.execute(Statement::from_sql_and_values(DbBackend::MySql, r#"CALL test_procedure(?, ?, @addedUser, @insertRowCount, @rows_affected);"#, [uuid.into(), email.into()])).await.unwrap();

println!("{:?}", exec) // this outputs, ExecResult { result: SqlxMySql(MySqlQueryResult { rows_affected: 0, last_insert_id: 0 }) }
```
Then i have a simple procedure like 
```
CREATE PROCEDURE `test_procedure` (IN uuid TEXT, IN email TEXT, OUT addedUser INT, OUT insertRowCount INT,
    OUT rows_affected INT -- tried this to see if it will work with the exec.rows_affected()
)
BEGIN
    SET @addedUser = 0;
    SET @insertRowCount = 0;
    
      // SQL STATEMENT TO INSERT USER HERE //
        SELECT ROW_COUNT() INTO @insertRowCount;
        SET @addedUser = 1;
        
        IF (@insertRowCount <> 1) THEN
	    SET @addedUser = 0;
        END IF;
        
	SET rows_affected = ROW_COUNT(); -- tried this also to see if exec.rows_affected() will be set
        -- SET rows_affected = @addedUser;  -- tried this as well.
     -- SELECT  @addedTransaction, @insertRowCount; // tried this as well to get the output 
END
```

When the program runs a user is created but exec.rows_affected() always returns zero, And i will be needing output params for some checks at application level. Please is this possible with seaorm or do i trust that if no error was returned then all the operations succeeded without the need for app level check?

Any advice or alternative suggestions is much appreciated 🙏🏾

**Comment by Greatchinex** [comment]
@tyt2y3 Please pardon me tagging you, But i need help on this 🙏🏾

**Comment by tyt2y3** [comment]
I wanted to help, but I couldn't see what the proceduce does and why it's needed in the first place?

**Comment by Greatchinex** [comment]
@tyt2y3 The actual procedure is quite long, Thats why i gave the earlier example i gave as just a simple use case to insert a user in the user's table. But i am pretty much using it for updating wallet balances and creating entries into a transactions table and i just want to have a single method that does that(So as not to have insert queries for transactions all over the application). The above code i shared works, Just that i don't know how to get output back from the procedure. Cause i will like to do some app level checks with the procedure. 

I currently am just flying with the fact that if no error was returned then everything went smoothly. I just need some thoughts and advice thanks.
